src/sgame/sg_main.cpp

```diff
@@ -2991,9 +2991,20 @@ void G_RunFrame( int levelTime )
 void G_PrepareEntityNetCode() {
 	// TODO: Allow ForEntities with empty template arguments.
 	gentity_t *oldEnt = &g_entities[0];
+	std::list<Entity*> specEntities;
+	// Prepare netcode for all non-specs first.
 	for (int i = 0; i < level.num_entities; i++, oldEnt++) {
 		if (oldEnt->entity) {
+			if (oldEnt->entity->Get<SpectatorClassComponent>()) {
+				specEntities.push_back(oldEnt->entity);
+				continue;
+			}
 			oldEnt->entity->PrepareNetCode();
 		}
 	}
+
+	// Prepare netcode for specs
+	for (Entity* e : specEntities) {
```

Comment:
Just use the `ForEntities` templated iterator here. The reason it's not done above is, as the TODO implies, you cannot currently iterate over *all* entities with it, just over a subset of all entities that has a given nonempty subset of components.

The syntax is like this:

    ForEntities<SpectatorComponent>([&](Entity& entitiy, SpectatorComponent& spectatorComponent){
        entity.PrepareNetcode();
    });